core/embed/rust/src/zcash/orchard_bridge.rs

```diff
@@ -0,0 +1,94 @@
+use core::convert::{TryFrom, TryInto};
+use core::ops::{Deref, DerefMut};
+
+use cstr_core::CStr;
+use f4jumble;
+use orchard::keys::{FullViewingKey, IncomingViewingKey, SpendingKey};
+
+use crate::error::Error;
+use crate::micropython::{
+    buffer::{Buffer, BufferMut},
+    obj::Obj,
+};
+use crate::util;
+
+/// Returns Orchard Full Viewing Key.
+fn get_orchard_fvk(sk: Obj) -> Result<FullViewingKey, Error> {
+    let sk: Buffer = sk.try_into()?;
+    let mut sk_bytes = [0u8; 32];
+    sk_bytes.copy_from_slice(sk.deref());
+
+    // SpendingKey is not valid with a negligible probability.
+    let sk = SpendingKey::from_bytes(sk_bytes);
+    // conversion from CtOption to Option
+    let sk: Option<SpendingKey> = sk.into();
+    let err_msg =
+        unsafe { CStr::from_bytes_with_nul_unchecked("Invalid Spending Key\0".as_bytes()) };
+    let sk = sk.ok_or(Error::ValueError(err_msg))?;
```

Comment:
`ok_or_else(|| ...)` would be more idiomatic here (so you don't needlessly create the error objects).